packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx

```diff
@@ -115,14 +144,51 @@ export const XRPLTransaction: FC<XRPLTxProps> = ({
     );
   };
 
+  const renderCurrency = (params: { title: string; value: Currency }) => {
+    const { title, value } = params;
+    const currency =
+      value.currency.length === 40 ? convertHexCurrencyString(value.currency) : value.currency;
```

Comment:
You don't need to do this checking, this check is done within: `convertHexCurrencyString`.
const currency = convertHexCurrencyString(value.currency); should be enough